pkg/server/environment_create.go

```diff
@@ -27,32 +24,31 @@ import (
 // @Success     201            {object} types.EnvironmentCreateResponse
 // @Router      /users/{login_name}/environments [post]
 func (s Server) environmentCreate(c *gin.Context) error {
-	it := c.GetString(ContextLoginName)
+	owner := c.GetString(ContextLoginName)
 
 	var req types.EnvironmentCreateRequest
 	if err := c.BindJSON(&req); err != nil {
 		return NewError(http.StatusInternalServerError, err, "gin.bind-json")
 	}
 
-	meta, err := image.FetchMetadata(c.Request.Context(), req.Spec.Image)
+	// Create the image in DB if it does not exist.
+	meta, err := s.ImageService.CreateImageIfNotExist(
+		c.Request.Context(), owner, req.Spec.Image)
 	if err != nil {
-		return NewError(http.StatusInternalServerError, err, "image.fetch-metadata")
+		return NewError(http.StatusInternalServerError,
+			err, "image-service.create-image")
 	}
-	var pglabel pgtype.JSONB
-	err = pglabel.Set(meta.Labels)
-	if err != nil {
-		return NewError(http.StatusInternalServerError, err, "pglabel.set")
-	}
-	_, err = s.Queries.CreateImageInfo(context.Background(),
-		query.CreateImageInfoParams{OwnerToken: it,
-			Name: meta.Name, Digest: meta.Digest, Created: meta.Created, Size: meta.Size, Labels: pglabel})
-	if err != nil {
-		return NewError(http.StatusInternalServerError, err, "db.create-image")
+	if meta == nil {
+		return NewError(http.StatusInternalServerError,
+			errors.New("meta is nil"), "image-service.create-image")
```

Comment:
This error message is not straightforward for end users.

Reply:
Do you mean meta is nil?

Reply:
Yes